TSG/Update/cau-debug-in-23h2.md

```diff
@@ -0,0 +1,263 @@
+
+- Common error codes and error strings
+
+    > CBS_E_NEW_SERVICING_STACK_REQUIRED - usually means an update was already installed and is pending reboot. 
+    > MOSETUP_E_SETUP_PLATFORM 
+    > ERROR_FILE_NOT_FOUND
+    > E_CABAPI_FDI_TARGET_FILE - a previously failed CAU run has exited without cleaning up.  
+    > ERROR_ACCESS_DENIED
+    > ERROR_WRITE_PROTECT
+    > MOSETUP_E_UA_NO_PACKAGES_TO_DOWNLOAD
+    > SEC_E_WRONG_PRINCIPAL
+    > WBEM_E_SHUTTING_DOWN
+    > E_DEVICE_ACCESS_DENIED
+    > RPC_S_CALL_FAILED
+    > WEB_E_INVALID_JSON_STRING
+    
+    > Rollback error 
+    > WinMgmt did not start error 
+    > Audi error 
+    > Node already down error 
+    > CIM Error 
+    > Clussvc did not start
+    > Drain failure
+    > Restart timeout reached
+    > GetPostRebootResult error
+
+- Installation errors: 
+
+    CAU may encounter errors while installing the updates on a cluster node, such as the not being able to expand installation files, or access to the files, or errors during install itself, etc. Plugin related errors – error codes will have enough details themselves. Ex: <mark>PluginReported_InstallFailure_AudiInstall_PayloadNotFound_0x80040406</mark>
+
+- Node drain failures: 
+    
+    CAU may fail to drain workloads prior to rebooting from nodes due to various reasons, such as the cluster storage being degraded and/or in repair, problems with VM live migration, etc. Collecting and looking into cluster logs will give a better indication on cause. Need to narrow down based on timestamps. Cluster logs live migration failure sample:
+
+    > 00004ad0.0000dda8::2024/06/13-20:00:50.248 WARN  [RES] Virtual Machine <Virtual Machine NODE002>: Live migration of 'Virtual Machine NODE002' failed.
+    >  
+    > Virtual machine migration operation for 'NODE002' failed at migration destination 'NODE002'. (Virtual machine ID D65EFFCA-1FE6-455E-8BA6-1B9C18C3C4C6) 
+    > 
+    > Failed to perform migration on virtual machine 'NODE002' because virtual machine migration limit '1' was reached, please wait for completion of an ongoing migration
+
+- How to recover and reset state? 
+
+    [Invoke-CauRun (ClusterAwareUpdating) | Microsoft Learn](https://learn.microsoft.com/en-us/powershell/module/clusterawareupdating/invoke-caurun?view=windowsserver2022-ps#example-3-recover-from-a-previous-updating-run-that-failed-on-the-specified-cluster) with ForceRecovery
+
+- Triggering runs directly using powershell (NOT RECOMMENDED)
+    
+    [Invoke-CauRun (ClusterAwareUpdating) | Microsoft Learn](https://learn.microsoft.com/en-us/powershell/module/clusterawareupdating/invoke-caurun?view=windowsserver2022-ps) and/or [Set-CauClusterRole (ClusterAwareUpdating) | Microsoft Learn](https://learn.microsoft.com/en-us/powershell/module/clusterawareupdating/set-cauclusterrole?view=windowsserver2022-ps#example-3-initiate-an-updating-run-on-the-specified-cluster). You can get the exact cmdlet to run from the previous `Get-CauReport`.
+
+# TSGs
```

Comment:
Please remove references to internal TSG information.

Reply:
This is public documentation